rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java

```diff
@@ -279,10 +278,12 @@ protected void prepareUndoLog(TableRecords beforeImage, TableRecords afterImage)
 
         TableRecords lockKeyRecords = sqlRecognizer.getSQLType() == SQLType.DELETE ? beforeImage : afterImage;
         String lockKeys = buildLockKey(lockKeyRecords);
-        connectionProxy.appendLockKey(lockKeys);
+        if (null != lockKeys) {
```

Comment:
maybe you can put it in the connectionProxy# appendLockKey function

Reply:
If you put code in `connectionProxy# appendLockKey`, it will still execute L284~L285.